backend/src/aportalserver/models/courses.py

```diff
@@ -0,0 +1,41 @@
+from aportalserver.models.sqlutils import get_conn
+
+class Course:
+    def __init__(self, course_id, conn=None):
+        if conn is None:
+            self.conn = get_conn()
+        else:
+            self.conn = conn
+        self.course_id = course_id
+
+    def select(self):
+        with self.conn.cursor() as cur:
+            cur.execute("SELECT * FROM courses WHERE course_id=%s" % (self.course_id,))
+            return cur.fetchone()
+
+    def retrieve_taken_courses(self):
+        with self.conn.cursor() as cur:
+            cur.execute("SELECT * FROM taken_courses NATURAL JOIN users WHERE course_id=%s" % (self.course_id,))
```

Comment:
1. Any reason you're using a natural join here instead of like maybe an inner join?
2. Are you trying to find all the courses that a given user has taken? If so, how is a user being used here?